### PR TITLE
Add RestPacket for Client and Server

### PR DESCRIPTION
**Pull request by PonyWarrior**
https://github.com/BlowaXD/ChickenAPI/issues/47

### CODE REVIEW

**Comment by Elendan** [comment]
Please fix the properties name : PlayerID -> PlayerId

**Comment by BlowaXD** [comment]
Your pull request is actually not acceptable as is, I didn't test having different packet serialization for a single header, i need to give it a look.

File: src/ChickenAPI/Packets/Game/Client/RestPacket.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿namespace ChickenAPI.Packets.Game.Client
+{
+    [PacketHeader("rest")]
+    public class RestPacket : PacketBase
+    {
+        [PacketIndex(0)]
+        public byte Unknown1 { get; set; } // always set to 1
+
+        [PacketIndex(1)]
+        public byte Unknown2 { get; set; } // always set to 1
```
Comment (BlowaXD) [inline src/ChickenAPI/Packets/Game/Client/RestPacket.cs]: It's EntityType (1 = Player / 2 = Npc / 3 Monster)

File: src/ChickenAPI/Packets/Game/Server/RestPacket.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿namespace ChickenAPI.Packets.Game.Server
+{
+    [PacketHeader("rest")]
+    public class RestPacket : PacketBase
+    {
+        [PacketIndex(0)]
+        public byte Unknown1 { get; set; } // always set to 1
```
Comment (BlowaXD) [inline src/ChickenAPI/Packets/Game/Server/RestPacket.cs]: EntityType here too

File: src/ChickenAPI/Packets/Game/Server/RestPacket.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+        [PacketIndex(1)]
+        public int PlayerID { get; set; }
```
Comment (BlowaXD) [inline src/ChickenAPI/Packets/Game/Server/RestPacket.cs]: EntityId (not the real one, it's CharacterId / MonsterId / NpcId...)

File: src/ChickenAPI/Packets/Game/Server/RestPacket.cs [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+        [PacketIndex(2)]
+        public byte Unknown2 { get; set; } // always set to 0
```
Comment (BlowaXD) [inline src/ChickenAPI/Packets/Game/Server/RestPacket.cs]: This property equals the following condition : IsSitting == true

File: src/ChickenAPI/Packets/Game/Client/RestPacket.cs [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+        [PacketIndex(2)]
+        public int PlayerId { get; set; }
```
Comment (BlowaXD) [inline src/ChickenAPI/Packets/Game/Client/RestPacket.cs]: EntityId (same as on SentPacket, CharacterId, MonsterId, NpcId..)

File: src/ChickenAPI/Packets/Game/Server/RestPacket.cs [hunk 6]
```diff
@@ -0,0 +1,16 @@
+﻿namespace ChickenAPI.Packets.Game.Server
+{
+    [PacketHeader("rest")]
+    public class RestPacket : PacketBase
+    {
+        [PacketIndex(0)]
+        public byte Unknown1 { get; set; } // always set to 1
+
+        [PacketIndex(1)]
+        public int PlayerId { get; set; }
```
Comment (BlowaXD) [inline src/ChickenAPI/Packets/Game/Server/RestPacket.cs]: EntityId (CharacterId, MonsterId, NpcId..)

File: src/ChickenAPI/Packets/Game/Server/RestPacket.cs [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+        [PacketIndex(2)]
+        public byte Unknown2 { get; set; } // always set to 0
```
Comment (BlowaXD) [inline src/ChickenAPI/Packets/Game/Server/RestPacket.cs]: This represents the following condition : Entity.IsSitting = true

**Comment by BlowaXD** [comment]
You should fix your pull request as discussed.
- Use Enums
- SubPackets

**Pull request closed without merging**